core/src/main/kotlin/com/strumenta/kolasu/parsing/Parsing.kt

```diff
@@ -238,18 +239,23 @@ fun Parser.injectErrorCollectorInParser(issues: MutableList<Issue>) {
     this.removeErrorListeners()
     this.addErrorListener(object : BaseErrorListener() {
         override fun syntaxError(
-            p0: Recognizer<*, *>?,
-            p1: Any?,
+            recognizer: Recognizer<*, *>?,
+            offendingSymbol: Any?,
             line: Int,
             charPositionInLine: Int,
             errorMessage: String?,
-            p5: RecognitionException?
+            recognitionException: RecognitionException?
         ) {
+            val startPoint = Point(line, charPositionInLine)
+            var endPoint = startPoint
+            if (offendingSymbol is CommonToken) {
+                endPoint = startPoint.plus(offendingSymbol.text)
```

Comment:
We have the extension method `Token.endPoint` (and `Token.position` that calls it) that take into account the EOF token corner case, so I suggest we use it here.